lib/stages/integrateHFBData.cpp

```diff
@@ -0,0 +1,228 @@
+#include <string>
+
+using std::string;
+
+#include "chimeMetadata.h"
+#include "gpsTime.h"
+#include "hfbMetadata.h"
+#include "integrateHFBData.hpp"
+
+using kotekan::bufferContainer;
+using kotekan::Config;
+using kotekan::Stage;
+
+REGISTER_KOTEKAN_STAGE(integrateHFBData);
+
+integrateHFBData::integrateHFBData(Config& config_, const string& unique_name,
+                                   bufferContainer& buffer_container) :
+    Stage(config_, unique_name, buffer_container, std::bind(&integrateHFBData::main_thread, this)) {
+
+    // Apply config.
+    _num_frb_total_beams = config.get<uint32_t>(unique_name, "num_frb_total_beams");
+    _num_frames_to_integrate =
+        config.get_default<uint32_t>(unique_name, "num_frames_to_integrate", 80);
+    _num_sub_freqs = config.get<uint32_t>(unique_name, "num_sub_freqs");
+    _samples_per_data_set = config.get<uint32_t>(unique_name, "samples_per_data_set");
+    _good_samples_threshold = config.get<float>(unique_name, "good_samples_threshold");
+
+    in_buf = get_buffer("hfb_input_buf");
+    register_consumer(in_buf, unique_name.c_str());
+
+    compressed_lost_samples_buf = get_buffer("compressed_lost_samples_buf");
+    register_consumer(compressed_lost_samples_buf, unique_name.c_str());
+
+    out_buf = get_buffer("hfb_output_buf");
+    register_producer(out_buf, unique_name.c_str());
+}
+
+integrateHFBData::~integrateHFBData() {}
+
+void integrateHFBData::initFirstFrame(float* input_data, float* sum_data,
+                                      const uint32_t in_buffer_ID) {
+
+    int64_t fpga_seq_num_start =
+        fpga_seq_num_end - (_num_frames_to_integrate - 1) * _samples_per_data_set;
+    memcpy(&sum_data[0], &input_data[0], _num_frb_total_beams * _num_sub_freqs * sizeof(float));
```

Comment:
```suggestion
    memcpy(sum_data, input_data, _num_frb_total_beams * _num_sub_freqs * sizeof(float));
```
If it complains then cast to `(void *)`, but there is no pointing getting the pointer to the first element of an array, when the array itself is that value.